mike/themes/material/js/version-select.js

```diff
@@ -1,49 +1,46 @@
 window.addEventListener("DOMContentLoaded", function() {
-  // This is a bit hacky. Figure out the base URL from a known CSS file the
-  // template refers to...
-  var ex = new RegExp("/?assets/fonts/material-icons.css$");
-  var sheet = document.querySelector('link[href$="material-icons.css"]');
-
-  var ABS_BASE_URL = sheet.href.replace(ex, "");
-  var CURRENT_VERSION = ABS_BASE_URL.split("/").pop();
-
+  var VERSION = window.location.pathname.split('/')[1];
+  
   function makeSelect(options, selected) {
-    var select = document.createElement("select");
-    select.classList.add("form-control");
-
-    options.forEach(function(i) {
-      var option = new Option(i.text, i.value, undefined,
-                              i.value === selected);
-      select.add(option);
-    });
-
-    return select;
+      var select = document.createElement("select");
+      select.classList.add("form-control");
+  
+      options.forEach(function(i) {
+          var option = new Option(i.text, "/" + i.value, undefined,
+                                  i.value === selected);
+          select.add(option);
+      });
+  
+      return select;
   }
-
+  
   var xhr = new XMLHttpRequest();
-  xhr.open("GET", ABS_BASE_URL + "/../versions.json");
+  // Obtain JSON listing all available versions
+  xhr.open("GET", window.location.origin + "/versions.json");
```

Comment:
As mentioned at the top, this won't work for most Github pages (of the form `https://user.github.io/project/1.0/...`).